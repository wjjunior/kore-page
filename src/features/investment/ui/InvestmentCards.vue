<template>
  <div class="flex space-x-6">
    <div class="w-[441px] flex-shrink-0">
      <div
        class="bg-white rounded-lg border border-blue-400 overflow-hidden h-full"
      >
        <div class="p-4">
          <div class="border-b border-blue-400 mb-4 pb-4">
            <h2
              class="text-xl font-bold text-primary-200 font-hanken-grotesk leading-6"
            >
              Offering Terms
            </h2>
          </div>

          <div v-if="loading" class="flex flex-col gap-4">
            <div v-for="i in 5" :key="i" class="animate-pulse">
              <div class="h-3 bg-gray-200 rounded mb-1 w-24"></div>
              <div class="h-4 bg-gray-300 rounded w-48"></div>
            </div>
          </div>

          <div v-else class="flex flex-col gap-4">
            <div v-for="term in offeringTerms" :key="term.label">
              <h3 class="text-xs font-medium mb-1 text-primary-200 leading-5">
                {{ term.label }}
              </h3>
              <p class="font-bold text-base leading-5 text-gray-6">
                {{ term.value }}
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>

    <div class="w-[441px] flex-shrink-0">
      <div
        class="bg-white rounded-lg border border-blue-400 overflow-hidden h-full"
      >
        <div class="p-4">
          <div class="border-b border-blue-400 mb-4 pb-4">
            <h2
              class="text-xl font-bold text-primary-200 font-hanken-grotesk leading-6"
            >
              Documents
            </h2>
          </div>

          <div v-if="loading" class="flex flex-col gap-4">
            <div v-for="i in 5" :key="i" class="animate-pulse">
              <div class="h-3 bg-gray-200 rounded mb-2 w-32"></div>
              <div class="flex items-center space-x-2">
                <div class="w-4 h-4 bg-gray-200 rounded"></div>
                <div class="h-4 bg-gray-300 rounded w-40"></div>
              </div>
            </div>
          </div>

          <div v-else class="flex flex-col gap-4">
            <div v-for="document in documents" :key="document.id">
              <h3
                class="text-xs font-medium text-primary-200 mb-2 leading-5 align-middle"
              >
                {{ document.title }}:
              </h3>
              <div class="flex items-center space-x-2">
                <FileIcon />
                <a
                  href="#"
                  class="text-base font-bold text-gray-6 leading-5 align-middle underline underline-offset-4"
                >
                  {{ document.filename }}
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { FileIcon } from "@/shared/ui";
import type { OfferingTerm, Document } from "@/shared/lib/types";

interface Props {
  offeringTerms: OfferingTerm[];
  documents: Document[];
  loading?: boolean;
}

withDefaults(defineProps<Props>(), {
  loading: false,
});
</script>
