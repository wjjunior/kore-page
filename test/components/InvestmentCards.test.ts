import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import InvestmentCards from "@/features/investment/ui/InvestmentCards.vue";
import { mockOfferingTerms, mockDocuments } from "../mocks";

const mountInvestmentCards = (
  props: {
    offeringTerms?: typeof mockOfferingTerms;
    documents?: typeof mockDocuments;
    loading?: boolean;
  } = {}
) => {
  const defaultProps = {
    offeringTerms: [],
    documents: [],
    loading: false,
    ...props,
  };

  return mount(InvestmentCards, {
    props: defaultProps,
  });
};

describe("InvestmentCards", () => {
  it("should display 'Offering Terms' and 'Documents' titles", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: mockOfferingTerms,
      documents: mockDocuments,
    });

    const titles = wrapper.findAll("h2");
    const titleTexts = titles.map((h2) => h2.text());

    expect(titleTexts).toContain("Offering Terms");
    expect(titleTexts).toContain("Documents");
  });

  it("should render offering terms data when not loading", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: mockOfferingTerms,
      loading: false,
    });

    mockOfferingTerms.forEach(({ label, value }) => {
      expect(wrapper.text()).toContain(label);
      expect(wrapper.text()).toContain(value);
    });
  });

  it("should render documents data when not loading", () => {
    const wrapper = mountInvestmentCards({
      documents: mockDocuments,
      loading: false,
    });

    mockDocuments.forEach(({ title, filename }) => {
      expect(wrapper.text()).toContain(title);
      expect(wrapper.text()).toContain(filename);
    });
  });

  it("should render document filenames as links", () => {
    const wrapper = mountInvestmentCards({
      documents: mockDocuments,
    });

    const links = wrapper.findAll("a");
    expect(links.length).toBe(mockDocuments.length);

    links.forEach((link, index) => {
      expect(link.text()).toContain(mockDocuments[index]?.filename);
    });
  });

  it("should not render offering terms when loading is true", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: mockOfferingTerms,
      loading: true,
    });

    mockOfferingTerms.forEach(({ label, value }) => {
      expect(wrapper.text()).not.toContain(label);
      expect(wrapper.text()).not.toContain(value);
    });
  });

  it("should not render documents when loading is true", () => {
    const wrapper = mountInvestmentCards({
      documents: mockDocuments,
      loading: true,
    });

    mockDocuments.forEach(({ title, filename }) => {
      expect(wrapper.text()).not.toContain(title);
      expect(wrapper.text()).not.toContain(filename);
    });
  });

  it("should show skeleton elements when loading is true", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: mockOfferingTerms,
      documents: mockDocuments,
      loading: true,
    });

    const skeletons = wrapper.findAll(".animate-pulse");
    expect(skeletons.length).toBeGreaterThan(0);
  });

  it("should render nothing for offeringTerms and documents when empty", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: [],
      documents: [],
    });

    expect(wrapper.findAll("h3").length).toBe(0);
    expect(wrapper.findAll("a").length).toBe(0);
  });

  it("should match snapshot", () => {
    const wrapper = mountInvestmentCards({
      offeringTerms: mockOfferingTerms,
      documents: mockDocuments,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
