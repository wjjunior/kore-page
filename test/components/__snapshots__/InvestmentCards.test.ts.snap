// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestmentCards > should match snapshot 1`] = `
"<div class="flex flex-col lg:flex-row lg:space-x-4 xl:space-x-6 space-y-6 lg:space-y-0">
  <div class="w-full lg:w-[400px] xl:w-[441px] lg:flex-shrink-0">
    <div class="bg-white rounded-lg border border-blue-400 overflow-hidden h-full">
      <div class="p-4">
        <div class="border-b border-blue-400 mb-4 pb-4">
          <h2 class="text-xl font-bold text-primary-200 font-hanken-grotesk leading-6"> Offering Terms </h2>
        </div>
        <div class="flex flex-col gap-4">
          <div>
            <h3 class="text-xs font-medium mb-1 text-primary-200 leading-5">Regulation</h3>
            <p class="font-bold text-base leading-5 text-gray-6">Regulation Crowdfunding (RegCF)</p>
          </div>
          <div>
            <h3 class="text-xs font-medium mb-1 text-primary-200 leading-5">Offering Type</h3>
            <p class="font-bold text-base leading-5 text-gray-6">Revenue Sharing Agreement</p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="w-full lg:w-[400px] xl:w-[441px] lg:flex-shrink-0">
    <div class="bg-white rounded-lg border border-blue-400 overflow-hidden h-full">
      <div class="p-4">
        <div class="border-b border-blue-400 mb-4 pb-4">
          <h2 class="text-xl font-bold text-primary-200 font-hanken-grotesk leading-6"> Documents </h2>
        </div>
        <div class="flex flex-col gap-4">
          <div>
            <h3 class="text-xs font-medium text-primary-200 mb-2 leading-5 align-middle">Form C: </h3>
            <div class="flex items-center space-x-2"><svg width="16" height="20" viewBox="0 0 16 20" fill="none" xmlns="http://www.w3.org/2000/svg" class="w-4 h-5 text-neutral-400">
                <path d="M15.903 6.586C15.854 6.48 15.793 6.379 15.707 6.293L9.707 0.293C9.621 0.207 9.52 0.146 9.414 0.0969999C9.384 0.0829999 9.352 0.0749999 9.32 0.0639999C9.236 0.0359999 9.15 0.018 9.061 0.013C9.04 0.011 9.021 0 9 0H2C0.897 0 0 0.897 0 2V18C0 19.103 0.897 20 2 20H14C15.103 20 16 19.103 16 18V7C16 6.979 15.989 6.96 15.987 6.938C15.982 6.849 15.965 6.763 15.936 6.679C15.926 6.647 15.917 6.616 15.903 6.586ZM12.586 6H10V3.414L12.586 6ZM2 18V2H8V7C8 7.553 8.447 8 9 8H14L14.002 18H2Z" fill="currentColor"></path>
                <path d="M4 10H12V12H4V10ZM4 14H12V16H4V14ZM4 6H6V8H4V6Z" fill="currentColor"></path>
              </svg><a href="#" class="text-base font-bold text-gray-6 leading-5 align-middle underline underline-offset-4 break-words">FileName_GoesHere.pdf</a></div>
          </div>
          <div>
            <h3 class="text-xs font-medium text-primary-200 mb-2 leading-5 align-middle">Custodian and Voting Agreement: </h3>
            <div class="flex items-center space-x-2"><svg width="16" height="20" viewBox="0 0 16 20" fill="none" xmlns="http://www.w3.org/2000/svg" class="w-4 h-5 text-neutral-400">
                <path d="M15.903 6.586C15.854 6.48 15.793 6.379 15.707 6.293L9.707 0.293C9.621 0.207 9.52 0.146 9.414 0.0969999C9.384 0.0829999 9.352 0.0749999 9.32 0.0639999C9.236 0.0359999 9.15 0.018 9.061 0.013C9.04 0.011 9.021 0 9 0H2C0.897 0 0 0.897 0 2V18C0 19.103 0.897 20 2 20H14C15.103 20 16 19.103 16 18V7C16 6.979 15.989 6.96 15.987 6.938C15.982 6.849 15.965 6.763 15.936 6.679C15.926 6.647 15.917 6.616 15.903 6.586ZM12.586 6H10V3.414L12.586 6ZM2 18V2H8V7C8 7.553 8.447 8 9 8H14L14.002 18H2Z" fill="currentColor"></path>
                <path d="M4 10H12V12H4V10ZM4 14H12V16H4V14ZM4 6H6V8H4V6Z" fill="currentColor"></path>
              </svg><a href="#" class="text-base font-bold text-gray-6 leading-5 align-middle underline underline-offset-4 break-words">FileName_GoesHere.pdf</a></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
